src/cleanupCommand/cleanupManager.ts

```diff
@@ -94,13 +98,18 @@ export class CleanupManager {
         sourcesToDelete: sourcesToDelete.length ? sourcesToDelete.map((source) => source.parameters.originDirectory) : [],
         ignoredSources: ignoredSources.length ? ignoredSources.map((source) => source.parameters.originDirectory) : [],
       });
-      const sourcesDirectories = this.getSourcesLocation(blackListFilteredBatch);
-      await this.sourcesProvider.deleteDiscretes(sourcesDirectories);
+      const expiredBatch = this.filterExpiredFailedTasks(blackListFilteredBatch, this.successCleanupDelayDays);
+
+      if (!this.disableSourcesCleanup) {
+        const sourcesDirectories = this.getSourcesLocation(expiredBatch);
+        await this.sourcesProvider.deleteDiscretes(sourcesDirectories);
```

Comment:
can we enter the "if (!this.disableSourcesCleanup) {" logic inside "sourcesProvider.deleteDiscretes" and prevent it's duplication?